pkg/alps/job.go

```diff
@@ -0,0 +1,179 @@
+package alps
+
+import (
+	"context"
+	"fmt"
+	"os/exec"
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/google/uuid"
+)
+
+type JobStatus string
+
+const (
+	Stopped JobStatus = "stopped"
+	Running JobStatus = "running"
+)
+
+// Job represents a running or stopped job and all related details
+type Job struct {
+	// Owner is any string, used to identify who started the job
+	Owner string
+
+	// ID is the UUIDv4 identifying the job. Use this to retrieve the job from the Hub.
+	ID string
+
+	// StartTime is set as soon as the job is started.
+	//
+	// StopTime will be zero until the job has stopped.
+	startTime time.Time
+	stopTime  time.Time
+
+	// ExitCode will be available after the job has stopped.
+	exitCode *int
+
+	// Status will be available as soon as the job is started
+	status JobStatus
+
+	// stdErr and StdOut will contain the redirected stdErr and Stdout from the job
+	stdErr *safeBuffer
+	stdOut *safeBuffer
+
+	command string
+	args    []string
+	mu      sync.RWMutex
+	cmd     *exec.Cmd
+	done    chan bool
+}
+
+// CommandString returns the entire command with arguments used to start the job
+func (j *Job) CommandString() string {
+	return j.command + strings.Join(j.args, " ")
+}
+
+func (j *Job) Status() JobStatus {
+	j.mu.RLock()
+	defer j.mu.RUnlock()
+	return j.status
+}
+
+func (j *Job) StartTime() time.Time {
+	j.mu.RLock()
+	defer j.mu.RUnlock()
+	return j.startTime
+}
+
+func (j *Job) StopTime() time.Time {
+	j.mu.RLock()
+	defer j.mu.RUnlock()
+	return j.stopTime
+}
+
+func (j *Job) ExitCode() int {
+	j.mu.RLock()
+	defer j.mu.RUnlock()
+	return *j.exitCode
```

Comment:
if ExitCode is called on `running` job this will panic.

Reply:
Modified to return an "ok" if the exitCode is actually available and 0/not ok if it is not.